test/unit/Optimize.ml

```diff
@@ -416,7 +413,11 @@ let%expect_test "do not inline recursive functions" =
 
       log_prob {
         {
-          FnReject__(g(53));
+          real inline_g_return_sym1__;
+          {
+            inline_g_return_sym1__ = (53 ^ 2);
+          }
+          FnReject__(inline_g_return_sym1__);
         }
```

Comment:
That check is totally broken in the optimizer. I'm not sure it matters (it doesn't try to inline infinitely, it just does one round), but if we really wanted to avoid it we would need to do another set of callgraph building/checking.

Reply:
IIRC, I did experiment with removing function declarations from the MIR when working on #1277 and this test "breaking" was the reason I decided not to bother.

But as far as I know, the optimizer does handle recursion sensibly so all we need is to change this test so that it checks something nontrivial.